Fixes #15422 - proper way to override puppet env field

ShimShtein opened the pull request:
Moved from defacing the view to `alias_method_chain` to the helper method.
Since Foreman is handling the hostgroup's inheritance, #6102 could be reverted.

jlsherrill commented:
Apologies for the delay.  I'm reviewing this now, the conflict is a small one.

jlsherrill commented:
This seems to solve http://projects.theforeman.org/issues/15580 also

jlsherrill commented on app/overrides/add_activation_keys_input.rb:
```diff
@@ -10,12 +10,12 @@
 Deface::Override.new(:virtual_path => "hostgroups/_form",
                      :name => "hostgroups_update_environments_select",
-                     :replace => 'erb[loud]:contains("select_f"):contains(":environment_id")',
+                     :insert_bottom => 'div#primary',
```
I think it makes more sense for content view and lifecycle environment to be just prior to the puppet environment selector, as your selection for those values can change the value of the puppet env

jlsherrill commented on app/views/overrides/activation_keys/_host_environment_select.html.erb:
```diff
@@ -34,18 +34,10 @@ end %>
-    :help_inline => %(<a id="reset_puppet_environment">#{_("Reset Puppet Environment to match selected Content View")}</a>).html_safe
-  } %>
-
-
```
Also with this change it doesn't seem to be 'auto-selecting' the matching puppet environment for the selected content view and lifecycle environment.  Was this an intentional change?  I'm not necessarily opposed to it as the 'reset puppet environment' link still works, but was curious if intentional.

ehelms commented:
@ShimShtein looks like this needs a rebase and comments addressed

ShimShtein commented:
Rebased, changed the selector to pull katello fields before puppet environment and fixed auto-select.

About this: I had to force snake casing for `data-content_puppet_match` attribute. Maybe it would be nicer to follow the convention and convert underscores to dashes? Although I don't think it should be part of this PR.

jlsherrill commented:
[test]

jlsherrill commented:
ACK thanks @ShimShtein !
